src/Microsoft.Framework.FileSystemGlobbing/Internal/MatcherContext.cs

```diff
@@ -74,7 +96,44 @@ private void Match(DirectoryInfoBase directory, string parentRelativePath)
             }
 
             // Request all the including and excluding patterns to pop their status stack.
-            PopFrame();
+            PopDirectory();
+        }
+
+        private void Predict()
+        {
```

Comment:
Please help to find a proper term. The original word used by @lodejard was `Declare`. The functions do is to declare what's the next path segment to be processed. May be `Declare` is indeed more accurate name. I'll change it.